server/routes/wishCards.js

```diff
@@ -411,48 +419,198 @@ router.post(
+      io.emit('unblock', {id: wishCardId});
+      clearTimeout(blockedWishcardsTimer[wishCardId])
+    }
+
+    res.status(200).send({
+      success: true,
+      error: null,
+    });
+  } catch (error) {
+    handleError(res, 400, error);
+  }
+});
+
+
+router.get('/status/:id', async (req, res) => {
+
+  try {
+    const wishCardId = req.params.id;
+
+    if (!req.session.user) return handleError(res, 400, 'User not found');
+
+    const user = await UserRepository.getUserByObjectId(req.session.user._id);
+    if (!user) return handleError(res, 400, 'User not found');
+
+    clearTimeout(blockedWishcardsTimer[wishCardId])
+    const wishCard = await WishCardRepository.getWishCardByObjectId(wishCardId);
+
+    queue.add({wishCardId,
+      userId: req.session.user._id,
+      url:wishCard.wishItemURL,
+      price: wishCard.wishItemPrice});
+
+    res.status(200).send({
+      success: true,
+      error: null,
+    });
 
-        if (moment(wishcardAlreadyLockedByUser.isLockedUntil) > moment()) {
-          return handleError(res, 400, 'You already have a locked wishcard.');
+  } catch (error) {
+    handleError(res, 400, error);
+  }
+
+});
+
+
+queue.process(async (job, done) => {
+
+  const {wishCardId, userId, url, price} = job.data;
+
+  try {
+
+    let isDonated = true;
+    const wishListArray = /registryID\.1=(.*?)&.*?registryItemID.1=(.*?)&/gm.exec(url);
+    if(wishListArray) {
+
+      const wishListId = wishListArray[1];
+      const itemId = wishListArray[2];
+
+      if (wishListId) {
+        const scrapeResponse = await scrapeList(`https://www.amazon.com/hz/wishlist/ls/${wishListId}` );
+
+        console.log(scrapeResponse)
+        if (!scrapeResponse) isDonated = false;
+
+        if (Object.keys(scrapeResponse).length > 0) {
+          isDonated = !JSON.stringify(scrapeResponse).includes(itemId);
         }
 
-        const lockedWishCard = await WishCardRepository.lockWishCard(wishCardId, user._id);
-        io.emit('block', { id: wishCardId, lockedUntil: lockedWishCard.isLockedUntil });
+        if(isDonated) {
+
+          const wishCard = await WishCardRepository.getWishCardByObjectId(wishCardId);
+
+          wishCard.isDonated = true;
+          wishCard.save();
+
+          await DonationsRepository.createNewDonation({
+            donationTo: wishCardId,
+            donationFrom: userId,
+            donationPrice: price,
+            donationConfirmed: true
+          });
+
+          io.emit('donated', {id: wishCardId, donatedBy: userId});
+          done(true);
+          return true;
+
+        }
+        io.emit('not_donated', {id: wishCardId, donatedBy: userId});
+        done(false);
 
-        res.status(200).send({
-          success: true,
-          error: null,
-        });
       }
-    } catch (error) {
-      handleError(res, 400, error);
+      done(false)
+
     }
-  },
-);
+    done(false);
+
+  } catch (error) {
+    console.log(error);
```

Comment:
instead of console statements you can use the log function from helper/logger.js

better log formatting and more clear output

Reply:
fixed